pkg/apis/core/validation/validation_test.go

```diff
@@ -18839,7 +18837,7 @@ func TestValidateResourceNames(t *testing.T) {
 		{"kubernetes.io/will/not/work/", false, ""},
 	}
 	for k, item := range table {
-		err := validateResourceName(item.input, field.NewPath("field"))
+		err := validateResourceName(core.ResourceName(item.input), field.NewPath("field"))
```

Comment:
again, ou can just make the struct field be `core.ResourceName` type

Reply:
We need to cast `core.ResourceName` in this test, either here or in L8833～18834😢